java/ingest/src/main/java/sleeper/ingest/job/IngestJobRunner.java

```diff
@@ -116,26 +116,23 @@ public IngestResult ingest(IngestJob job) throws InterruptedException, IteratorE
                         + ". This file will be ignored and will not be ingested.", pathString);
             }
         }
+        // Run the ingest
+        try (CloseableIterator<Record> recordIterator = new ConcatenatingIterator(inputIterators);
```

Comment:
The ParquetReaders won't be closed this way. It looks like that was the case already.

When the ParquetReaderIterator constructor gets called, it reads the first record from ParquetReader immediately. That opens a file, and if neither of them are closed that won't be closed.

The ConcatenatingIterator only closes the iterator that it was currently working with when it closed.

Reply:
This is probably a separate issue.

Reply:
I just noticed the ConcatenatingIterator takes a list of suppliers rather than a list of iterators. Because it's using suppliers, it looks like it will only actually open the ParquetReaders one at a time, and close each one when it's finished. It's quite confusing but I'm not sure what the alternative is.